pkg/trace/api/profiles.go

```diff
@@ -101,25 +94,11 @@ func errorHandler(err error) http.Handler {
 // The tags will be added as a header to all proxied requests.
 // For more details please see multiTransport.
 func newProfileProxy(transport http.RoundTripper, targets []*url.URL, keys []string, tags string) *httputil.ReverseProxy {
-	director := func(req *http.Request) {
-		req.Header.Set("Via", fmt.Sprintf("trace-agent %s", info.Version))
-		if _, ok := req.Header["User-Agent"]; !ok {
-			// explicitly disable User-Agent so it's not set to the default value
-			// that net/http gives it: Go-http-client/1.1
-			// See https://codereview.appspot.com/7532043
-			req.Header.Set("User-Agent", "")
-		}
-		containerID := req.Header.Get(headerContainerID)
-		if ctags := getContainerTags(containerID); ctags != "" {
-			req.Header.Set("X-Datadog-Container-Tags", ctags)
-		}
-		req.Header.Set("X-Datadog-Additional-Tags", tags)
-		metrics.Count("datadog.trace_agent.profile", 1, nil, 1)
-		// URL, Host and key are set in the transport for each outbound request
-	}
 	logger := logutil.NewThrottled(5, 10*time.Second) // limit to 5 messages every 10 seconds
+	// URL, Host and key are set in the transport for each outbound request
+	// headers are set in the director
 	return &httputil.ReverseProxy{
-		Director:  director,
+		Director:  func(req *http.Request) { setCommonHeaders(req.Header, tags) },
```

Comment:
```suggestion
		Director:  func(req *http.Request) {
				setCommonHeaders(req.Header, tags)
				// URL, Host and API Keys are set by the Transport implementation multiTransport.
		},
```
And we can remove the comment above.